frontend/Dockerfile

```diff
@@ -15,18 +15,18 @@ FROM docker.io/nginx:stable-alpine
 LABEL maintainer="Chevdor <chevdor@gmail.com>"
 LABEL description="Polkadot Telemetry frontend"
 
+# When we run the built docker image, the default command uses this
+# env var to tell the UI where to look to get feed info.
 ENV SUBSTRATE_TELEMETRY_URL=
 
 WORKDIR /usr/share/nginx/html
 
-#COPY --from=builder /opt/builder/env.sh /usr/bin/
-#RUN apk add --no-cache bash; chmod +x /usr/bin/env.sh
+COPY --from=builder /opt/builder/env.sh /usr/bin/
```

Comment:
Instead of running `env.sh` inside the container, we can just put the output of the script in a file and mount it inside the container. users must change this file (instead of the env variable) if they're going to make some changes.

Generally, the best practice is to avoid passing config files and scripts in the build step as changing the runtime configuration needs rebuilding the image. It's not also compatible with the Kubernetes logic where we have configmaps, secrets and env variables.

I totally agree that maintaining two duplicate nginx configuration (one for docker-compose and the other for the helm chart) is not ideal, but we can't have multiple deployments on different platforms with a single configuration file. We could have a separate docker-compose directory and put some basic configuration files there for docker-compose users.

Reply:
I will open a new issue with those k8s related topics as discussed with @jsdw. 

> Instead of running env.sh inside the container, we can just put the output of the script in a file and mount it inside the container. users must change this file (instead of the env variable) if they're going to make some changes.

I would recommend against that. That means users would have to write and tweak a JS file for the solely purpose of passing ENV vars... Not only this is a garanty for errors but that also does not make much sense.

> Generally, the best practice is to avoid passing config files and scripts in the build step as changing the runtime configuration needs rebuilding the image. It's not also compatible with the Kubernetes logic where we have configmaps, secrets and env variables.

I agree. The env-config.js file is not generated at build time, it is generated as run time, based on the ENV the user passes. When the user uses k8s, those can be passed as ConfigMaps, Secrets or ENV as you mentioned.

> I totally agree that maintaining two duplicate nginx configuration (one for docker-compose and the other for the helm chart) is not ideal, but we can't have multiple deployments on different platforms with a single configuration file. We could have a separate docker-compose directory and put some basic configuration files there for docker-compose users.

I will open the k8s related issue and we can dig into that in there.

Reply:
I created an [issue for the k8s topics](https://github.com/paritytech/substrate-telemetry/issues/370) that are not directly related to this PR.